File: src/models/plugins.ts
export type Mode = "Install" | "Uninstall";
export type PluginFormat = "VST3" | "CLAP" | "LV2" | "MOD Audio";
export type ModPlatform = "Duo" | "DuoX" | "Dwarf";
export type PluginFolders = {
  vst3Folder?: string;
  clapFolder?: string;
  lv2Folder?: string;
};

export type FetchPluginsResponse = Record<PluginFormat, string[]> & {
  modIsConnected?: boolean;
};
export type SelectedPlugins = Record<PluginFormat, string[]>;

File: src-tauri/src/models/plugins.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Default)]
pub struct ModPlugins {
    #[serde(rename = "Duo")]
    pub duo: Vec<String>,
    #[serde(rename = "DuoX")]
    pub duo_x: Vec<String>,
    #[serde(rename = "Dwarf")]
    pub dwarf: Vec<String>,
}

#[derive(Serialize, Deserialize)]
pub struct PluginsConfig {
    #[serde(rename = "VST3")]
    pub vst3: Vec<String>,
    #[serde(rename = "CLAP")]
    pub clap: Vec<String>,
    #[serde(rename = "MOD Audio")]
    pub mod_audio: ModPlugins,
}

#[derive(Serialize, Deserialize, Default)]
pub struct GetPluginsResponse {
    #[serde(rename = "VST3")]
    pub vst3: Vec<String>,
    #[serde(rename = "CLAP")]
    pub clap: Vec<String>,
    #[serde(rename = "MOD Audio")]
    pub mod_audio: ModPlugins,
    #[serde(rename = "modIsConnected")]
    pub mod_is_connected: bool,
}

#[derive(Serialize, Deserialize, Default)]
pub struct SelectedPlugins {
    #[serde(rename = "VST3")]
    pub vst3: Vec<String>,
    #[serde(rename = "CLAP")]
    pub clap: Vec<String>,
    #[serde(rename = "MOD Audio")]
    pub mod_audio: Vec<String>,
}
